### PR TITLE
Updated streaming args for StreamingDataset subclasses

### PR DESCRIPTION
**Pull request by snarayan21**
Previously, the constructor for `StreamingFinetuningDataset` was not updated with the correct streaming args, but the `build_finetuning_dataloader` function that called it had some updated streaming args. This was causing errors upon dataset instantiation. This has been addressed.

Additionally, the new streaming args `batching_method` and `sampling_granularity` have been added to `StreamingFinetuningDataset` and `StreamingTextDataset`.

### CODE REVIEW

**Comment by dakinggg** [comment]
@snarayan21 do you need to update the pin in `setup.py`? This stuff only works with latest streaming, right?

File: llmfoundry/data/text_data.py [hunk 1]
```diff
@@ -66,7 +66,14 @@ class StreamingTextDataset(StreamingDataset):
         shuffle_algo (str): Which shuffling algorithm to use. Defaults to ``py1b``.
         shuffle_seed (int): Seed for Deterministic data shuffling. Defaults to ``9176``.
         shuffle_block_size (int): Unit of shuffle. Defaults to ``1 << 18``.
-        sampling_method (str): Which sampling method to use, either ``balanced`` or ``fixed``. Defaults to ``balanced``.
+        sampling_method (str): Which sampling method to use, either ``balanced`` or ``fixed``.
```
Comment (dakinggg) [inline llmfoundry/data/text_data.py:69]: to double check, the defaults here are the same as the implementation prior to these new args, right?

Comment (snarayan21) [reply llmfoundry/data/text_data.py:69]: Yepo

**Review by dakinggg — COMMENTED** [review]
Thanks @snarayan21! Going to run regression tests just to make sure the upgrade doesn't break something unexpected.

**Comment by boomanaiden154** [comment]
Thanks for the patch! Probably should've checked the constructors in `Streaming` a little bit more closely.

**Review by dakinggg — APPROVED** [review]

**Pull request merged**